sdk/storage/storage-blob/src/pollers/BlobStartCopyFromUrlPoller.ts

```diff
@@ -0,0 +1,191 @@
+import { delay } from "@azure/core-http";
+import { PollOperation, PollOperationState, Poller } from "@azure/core-lro";
+import { BlobStartCopyFromURLResponse } from "../generated/src/models";
+import { BlobClient, BlobStartCopyFromURLOptions } from "../BlobClient";
+
+/**
+ * Defines the operations needed from a BlobClient.
+ */
+export type CopyPollerBlobClient = Pick<BlobClient, "abortCopyFromURL" | "getProperties"> & {
+  startCopyFromURL(
+    copySource: string,
+    options?: BlobStartCopyFromURLOptions
+  ): Promise<BlobStartCopyFromURLResponse>;
+};
+
+/**
+ * The state used by the PollOperation.
+ */
+export interface BlobBeginCopyFromUrlPollState
+  extends PollOperationState<BlobStartCopyFromURLResponse> {
+  blobClient: CopyPollerBlobClient;
+  copyId?: string;
+  copyProgress?: string;
+  copySource: string;
+  startCopyFromURLOptions?: BlobStartCopyFromURLOptions;
+}
+
+/**
+ * The PollOperation responsible for:
+ *  - performing the initial startCopyFromURL
+ *  - checking the copy status via getProperties
+ *  - cancellation via abortCopyFromURL
+ */
+export interface BlobBeginCopyFromURLPollOperation
+  extends PollOperation<BlobBeginCopyFromUrlPollState, BlobStartCopyFromURLResponse> {}
+
+/**
+ *
+ */
+export interface BlobBeginCopyFromUrlPollerOptions {
+  blobClient: CopyPollerBlobClient;
+  copySource: string;
+  intervalInMs?: number;
+  onProgress?: (state: BlobBeginCopyFromUrlPollState) => void;
+  resumeFrom?: string;
+  startCopyFromURLOptions?: BlobStartCopyFromURLOptions;
+}
+
+/**
+ *
+ */
+export class BlobBeginCopyFromUrlPoller extends Poller<
+  BlobBeginCopyFromUrlPollState,
+  BlobStartCopyFromURLResponse
+> {
+  public intervalInMs: number;
+
+  constructor(options: BlobBeginCopyFromUrlPollerOptions) {
+    const {
+      blobClient,
+      copySource,
+      intervalInMs = 15000,
```

Comment:
Honestly... from my own testing it appeared to take ~12-15 seconds for copy progress to be updated. By no means a definitive value, but I didn't find anything from the service to suggest a different value, and it's configurable by the user.

Reply:
Sometimes for tiny blob copy, it could be done within seconds. Storage explorer uses copy to implement rename operation too. Waiting 15s for this kind of scenario sounds not good.

Reply:
I was trying to find a balance between making too many calls and not enough. I did notice that blob copies within the same storage account appear instant regardless of the blob size, and in this case the poller completes right away (this is because startCopyrFromURL will have a copyStatus of success). This value is configurable by the user as well, so if they want it to poll every second they can do so.

That said I'll trust your expertise on this so if the above note on copying within a storage account doesn't alleviate your concerns I'll reduce this to 2 seconds.

Reply:
Ok. I don't have a good number either. Customer shoud aware the default value, or way to customzie it.
Can you update description for beginCopyFromURL JSDoc about the default poll internval value? Or add example in JSDoc about how to customize this value.

Reply:
Done!